### PR TITLE
feat(api) add missing step to registration workflow

### PR DESCRIPTION
**Pull request by prmths128**
This is one of the last missing pieces to allow registering using the auth server.
The last piece if you ignore issues with other components.

=> needs a change to organization registration to work. There's an exception there (api_key check).
might be some other adjustments required (=> issue sqlalchemy update)
==> second commit fixes the issues (altough we'll have to review api management later, I just sidestep the previous behaviour)

Note: the new sqlalchemy query is more compatible with sqlalchemy 2.0

### CODE REVIEW

**Review by inimaz — CHANGES_REQUESTED** [review]

File: carbonserver/carbonserver/api/services/signup_service.py [hunk 1]
```diff
@@ -72,3 +81,26 @@ def new_user_setup(self, user: User) -> User:
...
+            self._user_repository.get_user_by_id(id_token["sub"])
+        except HTTPException as e:
+            if e.status_code == 404:
+                if not create:
+                    LOGGER.error("Authenticated user not found")
+                    raise
+                LOGGER.info("Authenticated user not found. Creating.")
+                name = id_token.get("fields", {}).get("name")
+
+                new_user = self._user_repository.create_user(
```
[the remark this answers is not in the document]
Comment (prmths128) [reply carbonserver/carbonserver/api/services/signup_service.py]: Ah, yes !
I just noticed that too with a DB error

[the remark this answers is not in the document]
Comment (prmths128) [reply carbonserver/carbonserver/api/services/signup_service.py]: fixed

**Review by inimaz — APPROVED** [review]
Looks good! Thanks as well for the fix on the `subscribe_user_to_org` method

**Review by SaboniAmine — APPROVED** [review]
That's great! Will it be possible to deploy this, in a near future ? This might help front end integration :-)

**Pull request merged**